### PR TITLE
Re-add helm chart under charts/metallb

### PR DESCRIPTION
**Pull request by gclawes**
Create a new helm chart integrating features of the previous chart, as
well as stable/metallb and bitnami/metallb.

Features:
* Support new environment variables in speaker
* Use PodMonitor instead of ServiceMonitor (no need to create Services)
* Create PrometheusRule to detect stale config and config-not-loaded
* Configurable PrometheusRule alerts for address pool exhaustion
* Support MetalLB controller creating memberlist secret
* Standardize labels using template helper
* Create config-watcher and pod-lister roles
* OPA/rego based chart output validation
* JSON Schema values validation

Fixes metallb/metallb#653

### CODE REVIEW

**Comment by russellb** [comment]
I don't know helm, so won't be much help in reviewing this, but one question:

Do you have any thoughts about testing this in CI?  If this is included with metallb, I feel like we should have a CI job that validates the helm charts so we're sure they work with each release that gets tagged.  An alternative would be to put it in its own repo so it can be tagged and released independently and documented for which releases it has been tested against.

Also, I know this is still marked as a draft, but could you also include some docs?

**Comment by gclawes** [comment]
Yeah, I can come up with some tests to validate that it produces valid k8s configs.  I also intend to add JSONSchema validations for the values file, once I'm confident nothing else needs to change for that.  Helm also has test that can be run against a running cluster (if we can get minikube working in CI that would be sufficient): https://helm.sh/docs/topics/chart_tests/.

In the long run I think it makes sense to release chart updates with metallb releases, since changes to how metallb is configured must be reflected in the chart.  In my experience maintaining a separate repo for a chart can become difficult.

Docs are my next step, just wanted to get the draft out there so people could see the template code.

**Comment by russellb** [comment]
There's been some recent discussion about getting a functional CI job based on `kind` working, so we could probably do a variant of that for the helm based install later

**Comment by gclawes** [comment]
I'm working on documentation.  What are your thoughts on hosting the helm chart on metallb.universe.tf?  What should the path for the chart repo be?  The other option would be github release artifacts/pages, helm has a tool and associated GitHub action to do this: https://github.com/helm/chart-releaser https://github.com/helm/chart-releaser-action

**Comment by daxmc99** [comment]
I like this change and I believe this should be nested under the project. 
Thanks for the change! Let us know if you need help or have questions!

**Comment by billimek** [comment]
Like @gclawes suggests, there is an established and demonstrated pattern of leveraging the helm chart testing action as well as the helm chart releaser action to make the automated linting, testing (via `kind`), and releasing charts very simple:

* [Chat Testing Action](https://github.com/helm/chart-testing-action)
* [Chart Releaser Action](https://github.com/helm/chart-releaser-action)
* [Example repo with it all tied-together](https://github.com/helm/charts-repo-actions-demo)

**Comment by gclawes** [comment]
I've updated a GitHub Actions workflow that runs the chart-testing lint job.  I'll add the chart test job once I get tests written.

**Comment by gclawes** [comment]
This is ready for review.

Here's a gist with the templated output of the chart with the default values: https://gist.github.com/gclawes/4b2f4a1bfd64e53bedc1e3700c2e56d6

Once the 0.9.4 release is made I'll rebase this branch and squash most of the commits together.

**Comment by gclawes** [comment]
The one remaining concern I have is related to the `secretgen` Job to generate the memberlist secret.  As it is right now, it will not work in air-gapped clusters because it runs a standard `alpine` image and installs `openssl`/`kubectl` on the fly:
```
            set -ex

            # install openssl and kubectl
            echo "http://dl-cdn.alpinelinux.org/alpine/edge/testing" >> /etc/apk/repositories
            apk --no-cache add openssl kubectl

            # generate a random string with no newlines
            # store in file secret does not leak to logs
            openssl rand -base64 128 > /tmp/secret

            # create secret
            kubectl \
              create secret generic metallb-memberlist \
              --from-file=secretkey=/tmp/secret
```

I can overcome this by adding a new image called `metallb/secretgen`, but that'd be more invasive to the `tasks.py`, it would involve creating new tasks for building an image without a metallb binary in it (all the current tasks assume a go binary is built with each image).

Ultimately I think we want the secret generation to be handled by the controller, but we may still want a Job to clean up the secret on uninstall of the chart, so an extra image may be warranted regardless.

**Comment by johananl** [comment]
@gclawes `v0.9.4` is now released :slightly_smiling_face: - FYI.

**Comment by shaneutt** [comment]
> @gclawes `v0.9.4` is now released slightly_smiling_face - FYI.

Sorry to bother you @johananl and @gclawes but I simply wanted to check something:

This appears to be targeting the `v0.10.0` milestone, but that milestone has no due date associated with it.

My organization has been monitoring this PR as we're intending to switch to this helm chart from the soon to be deprecated one at `helm/charts`, do you have any kind of even very rough estimate you can provide on when you think this chart might be merged and ready for general usage so that I can set some rough expectations back at my org?

Thank you for your time!

**Comment by gclawes** [comment]
@johananl working on the rebase this week.

@shaneutt I do not have a date for v0.10.0, though we do want to get it out as soon as possible (I would guess on the order of weeks, not months).  Others may have a better estimate.

**Comment by champtar** [comment]
@gclawes have you tried `helm upgrade` from the bitnami chart ? not a blocker but nice to have

**Comment by gclawes** [comment]
@champtar I have not.  Upgrading from the bitnami chart will likely not work, as the values and internal mechanics of this chart are sufficiently different.  I guess in theory it might work, but I'm unsure.

File: charts/metallb/templates/speaker.yaml
Comment (champtar) [inline charts/metallb/templates/speaker.yaml:57]: We need a "enable memberlist" boolean, set to true by default, that if set to false remove all the METALLB_ML*
We've had pretty good results with memberlist but BGP only user don't need it, and it's nice to have it for debug

File: charts/metallb/templates/rbac.yaml
Comment (champtar) [inline charts/metallb/templates/rbac.yaml:21]: I see you use printf here and template later, I prefer template, it's easier to read

Comment (champtar) [reply charts/metallb/templates/rbac.yaml:21]: `["{{ template "metallb.fullname" . }}-controller"]`

File: charts/metallb/templates/NOTES.txt
Comment (champtar) [inline charts/metallb/templates/NOTES.txt:6]: maybe print the inline config ?

Also if someone changed the ConfigMap directly and the next helm upgrade "revert" the config, is there a way to print the old and new config ?

It's more a limitation of Helm but having to type the repo and version just to change the current config without upgrading is not user friendly at all, so not a fan of configInline when `kubectl edit` is all we need ;)

**Review by champtar — COMMENTED** [review]
Some small things, still need to test it

File: charts/metallb/Chart.yaml
Comment (gclawes) [reply charts/metallb/Chart.yaml:6]: Good eye, I'll fix this.

File: charts/metallb/README.md
Comment (gclawes) [reply charts/metallb/README.md:5]: It can be automated in the version bump task, I'll add that

File: charts/metallb/values.yaml
Comment (gclawes) [reply charts/metallb/values.yaml:124]: When it is empty, it defaults to the value of `appVersion` in `Chart.yaml`.  This is done by this line:
```
image: {{ .Values.controller.image.repository }}:{{ .Values.controller.image.tag | default .Chart.AppVersion }}
```

Which was part of the default skeleton template provided by `helm create` in helm v3.

File: charts/metallb/templates/speaker.yaml
Comment (gclawes) [reply charts/metallb/templates/speaker.yaml:57]: I'll add that, thanks.  Will memberlist just not be enabled if the `METALLB_ML*` variables aren't present?

The switch will be in `.Values.metallb.speaker.memberlist.enabled` (boolean).

File: charts/metallb/templates/rbac.yaml
Comment (gclawes) [reply charts/metallb/templates/rbac.yaml:21]: Thanks for spotting the inconsistency, I'll fix it.

File: charts/metallb/templates/NOTES.txt
Comment (gclawes) [reply charts/metallb/templates/NOTES.txt:6]: I can display the inline config.  I don't think there's a way to get diffs between releases/upgrades.

File: charts/metallb/values.yaml
Comment (champtar) [reply charts/metallb/values.yaml:124]: I was talking about the pullPolicy, not the tag ;)

Comment (gclawes) [reply charts/metallb/values.yaml:124]: Ahh, gotcha.  Yeah, I can make that empty.  I think it came as part of the `helm create` boilerplate.

File: charts/metallb/templates/NOTES.txt
Comment (ilhaan) [reply charts/metallb/templates/NOTES.txt:6]: So [helm-diff](https://github.com/databus23/helm-diff) is a plugin that allows you to check diffs between releases/upgrades. There is talk of this plugin being merged into helm core [here](https://github.com/databus23/helm-diff/issues/211). 

However, I don't think this chart has to worry about manual changes being reverted upon a helm upgrade. I think it is well understood that manual changes won't persist if helm is part of your workflow and you have not updated helm upgrade params.

File: charts/metallb/templates/speaker.yaml
Comment (champtar) [reply charts/metallb/templates/speaker.yaml:57]: > Will memberlist just not be enabled if the METALLB_ML* variables aren't present?

yes
https://github.com/metallb/metallb/blob/main/internal/speakerlist/speakerlist.go#L54

File: charts/metallb/templates/NOTES.txt
Comment (champtar) [reply charts/metallb/templates/NOTES.txt:6]: > I think it is well understood that manual changes won't persist if helm is part of your workflow and you have not updated helm upgrade params.

you assume people know how things are working and are not following blindly procedures and first google reponse :P

I agree this is not the job of the chart, but it would have been nice. I'm thinking about the message displayed by apt when default config changes, do you want to keep local or maintainer version, see a diff, edit, ..., this is really a good UX IMO.

Comment (gclawes) [reply charts/metallb/templates/NOTES.txt:6]: I think helm diff is a reasonable solution, especially when combined with tools like [Praqma/helmsman](https://github.com/Praqma/helmsman).

File: charts/metallb/README.md
Comment (gclawes) [reply charts/metallb/README.md:5]: Ah, forgot I already did this:
```
run("perl -pi -e 's,^Current chart version is: .*,Current chart version is: `{}`,g' charts/metallb/README.md".format(version), echo=True)
```

File: charts/metallb/values.yaml
Comment (gclawes) [reply charts/metallb/values.yaml:135]: Agreed, this should be up to cluster operators, we can't make assumptions on their behalf.

File: charts/metallb/values.yaml
Comment (gclawes) [reply charts/metallb/values.yaml]: Made optional and updated values.schema.json to allow null values

File: charts/metallb/templates/rbac.yaml
Comment (gclawes) [reply charts/metallb/templates/rbac.yaml]: Fixed.

Comment (gclawes) [reply charts/metallb/templates/rbac.yaml]: Fixed

File: charts/metallb/templates/NOTES.txt
Comment (gclawes) [reply charts/metallb/templates/NOTES.txt]: Updated NOTES.txt template

**Comment by gclawes** [comment]
I've squashed the commit down to 5 commits.  If anyone things it should be more compact I can do that.

**Comment by champtar** [comment]
Seems good to me, you are just missing `.Values.metallb.speaker.memberlist.enabled`

File: charts/metallb/templates/psp.yaml
Comment (champtar) [inline charts/metallb/templates/psp.yaml:86]: See #764, `SYS_ADMIN` and `NET_ADMIN` must be remove IMO

**Comment by champtar** [comment]
Just tried upgrade from bitnami chart:
```
Error: UPGRADE FAILED: error validating "": error validating data: [ValidationError(Role.rules[0]): unknown field "resources:" in io.k8s.api.rbac.v1.PolicyRule, ValidationError(Role.rules[1]): unknown field "resources:" in io.k8s.api.rbac.v1.PolicyRule]
```

File: charts/metallb/templates/rbac.yaml
Comment (champtar) [inline charts/metallb/templates/rbac.yaml:76]: typo

File: charts/metallb/templates/rbac.yaml
Comment (champtar) [inline charts/metallb/templates/rbac.yaml:79]: typo

**Review by champtar — CHANGES_REQUESTED** [review]

**Comment by champtar** [comment]
Upgrade from bitnami to your chart seems to work \o/
```
# helm install metallb --namespace metallb-system bitnami/metallb --version 0.1.28 --set existingConfigMap=config
...
# helm upgrade metallb --namespace metallb-system metallb --version 0.9.4 --set existingConfigMap=config
Release "metallb" has been upgraded. Happy Helming!
NAME: metallb
LAST DEPLOYED: Sat Oct 24 21:11:26 2020
NAMESPACE: metallb-system
STATUS: deployed
REVISION: 2
```

File: charts/metallb/templates/rbac.yaml
Comment (gclawes) [reply charts/metallb/templates/rbac.yaml:76]: good eye!

**Comment by gclawes** [comment]
I wonder if it will work by transplanting the contents of `configInLine` instead of referencing an existing config map.

File: charts/metallb/templates/psp.yaml
Comment (gclawes) [reply charts/metallb/templates/psp.yaml:86]: I'll wait for input on #764 before updating this.  If this PR merges before that one, we can update it there.

File: charts/metallb/templates/rbac.yaml
Comment (gclawes) [reply charts/metallb/templates/rbac.yaml]: Fixed.

File: charts/metallb/templates/rbac.yaml
Comment (gclawes) [reply charts/metallb/templates/rbac.yaml]: Fixed.

File: charts/metallb/templates/speaker.yaml
Comment (gclawes) [reply charts/metallb/templates/speaker.yaml:60]: Added switch to disable these vars:

speaker - METALLB_ML_BIND_ADDR METALLB_ML_LABELS METALLB_ML_BIND_PORT METALLB_ML_SECRET_KEY
controller - METALLB_ML_SECRET_NAME

**Comment by lexfrei** [comment]
Is it ready to test with 0.9.5 on arm64?

**Comment by gclawes** [comment]
@lexfrei you can test with a few caveats:

- It won't be deployed to the github pages helm repo until this PR is merged, so you will have to install from the git repository.
- This depends on #747 to create the memberlist secret, until that is merged and 0.10.0 is released, you will have to follow the procedure to create the secret manually here: https://metallb.universe.tf/installation/#installation-by-manifest

**Comment by lexfrei** [comment]
Can you sync your config map syntax with [bitnami's one](https://github.com/bitnami/charts/blob/master/bitnami/metallb/templates/configmap.yaml)?
It can simplify the migration for bitnami's users.

**Comment by gclawes** [comment]
What specifically is preventing migration now?

**Comment by lexfrei** [comment]
@gclawes hm. Can't reproduce the unmarshal error now. Looks like my own typo.

Now I can see this `null` for my config, but all works fine, and configmap created properly.

```
NOTES:
MetalLB is now running in the cluster.
LoadBalancer Services in your cluster are now available on the IPs you
defined in MetalLB's configuration:

null

To see IP assignments, try `kubectl get services`.
```

**Comment by gclawes** [comment]
I see that in mine, I'll figure out what's going on and fix it:
```
2020-11-15 11:48:55 NOTICE: Install release [ metallb ] version [ 0.9.5 ] in namespace [ metallb-system ]
2020-11-15 11:48:59 NOTICE: Release "metallb" does not exist. Installing it now.
NAME: metallb
LAST DEPLOYED: Sun Nov 15 11:48:56 2020
NAMESPACE: metallb-system
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
MetalLB is now running in the cluster.
LoadBalancer Services in your cluster are now available on the IPs you
defined in MetalLB's configuration:

null

To see IP assignments, try `kubectl get services`.

2020-11-15 11:48:59 NOTICE: Finished: Install release [ metallb ] version [ 0.9.5 ] in namespace [ metallb-system ]
```

**Comment by gclawes** [comment]
Fixed rendering the config:
```
NOTES:
MetalLB is now running in the cluster.
LoadBalancer Services in your cluster are now available on the IPs you
defined in MetalLB's configuration:

config:
  address-pools:
  - addresses:
    - 10.102.0.0/16
    name: default
    protocol: bgp
  peers:
  - my-asn: 65003
    peer-address: 127.0.0.1
    peer-asn: 65002

To see IP assignments, try `kubectl get services`.
```

**Review by daxmc99 — APPROVED** [review]

**Comment by johananl** [comment]
Thanks a lot for the effort @gclawes! I'm currently reviewing and will update as soon as I can.

File: .circleci/config.yml
Comment (daxmc99) [inline .circleci/config.yml:21]: This appears to be fixed 
https://github.com/helm/helm/issues/8596

File: charts/metallb/templates/rbac.yaml
Comment (daxmc99) [inline charts/metallb/templates/rbac.yaml:5]: I get a `helm lint` error here 
`"test-release-metallb:controller": invalid metadata name, must match regex ^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$ and the length must not be longer than 253`

File: charts/metallb/templates/rbac.yaml
Comment (daxmc99) [inline charts/metallb/templates/rbac.yaml:28]: Same lint error

File: charts/metallb/templates/rbac.yaml
Comment (daxmc99) [inline charts/metallb/templates/rbac.yaml:107]: Same helm lint error

**Review by daxmc99 — CHANGES_REQUESTED** [review]
Thank you so much again @gclawes! 
I'm sorry this has been open for so long. I left some comments in the PR.

File: charts/metallb/templates/rbac.yaml
Comment (gclawes) [reply charts/metallb/templates/rbac.yaml:5]: Interesting, ClusterRole names with `:` are definitely valid:
```
$ kubectl get clusterrole | grep system | head -1
system:discovery                                                       2021-01-01T17:10:36Z
```

Comment (gclawes) [reply charts/metallb/templates/rbac.yaml:5]: Ah, I left this note in `.circleci/config.yml`:
``` 20   # FIXME: this is pinned to v3.0.0 because subsequent versions of ct pull in new versions of helm, which are subject
 21   #        to this bug: https://github.com/helm/helm/issues/8596, specifically for rbac object names with ':' in them
 22   helm-lint:
 23     working_directory: /repo
 24     docker:
 25       - image: quay.io/helmpack/chart-testing:v3.0.0
```

File: .circleci/config.yml
Comment (gclawes) [reply .circleci/config.yml:21]: Still appears to be a problem: https://github.com/helm/helm/issues/8835

Comment (daxmc99) [reply .circleci/config.yml:21]: Ahh okay, can we update the comment?

File: charts/metallb/templates/rbac.yaml
Comment (daxmc99) [reply charts/metallb/templates/rbac.yaml:5]: I am! This clear that up

Comment (gclawes) [reply charts/metallb/templates/rbac.yaml:5]: It appears they fixed the issue in helm/helm#8596 about deprecated API versions, but not the `:` issue.  I'll update the note in `.circleci/config.yml`

File: .circleci/config.yml
Comment (gclawes) [reply .circleci/config.yml:21]: Yeah, I'll do that.

File: charts/metallb/templates/rbac.yaml
Comment (gclawes) [reply charts/metallb/templates/rbac.yaml:5]: Fixed, adjusted the comment about this in `.circleci/config.yml` to point to the new upstream issue for `:` in name.

**Comment by gclawes** [comment]
@daxmc99 do these changes look good?

**Review by daxmc99 — APPROVED** [review]
LGTM! 
These changes look good to me. Also thank you for updating the release process to update the chart version.

File: .circleci/config.yml
Comment (daxmc99) [inline .circleci/config.yml:39]: Very small nit: This PR seems to mix CircleCI additions and Github Actions. 
Is there a reason we cannot use only CircleCI or Github Actions?

Comment (gclawes) [reply .circleci/config.yml:39]: Sorry for the late response.  The GitHub Action is a convenient pre-made way to publish the Helm chart in GitHub Pages.  The only reason to use it is to avoid porting that same logic to CircleCI.

File: charts/metallb/templates/controller.yaml
Comment (champtar) [inline charts/metallb/templates/controller.yaml:46]: not needed, this is already the default

File: charts/metallb/templates/controller.yaml
Comment (champtar) [inline charts/metallb/templates/controller.yaml:52]: Not required anymore (remove it)

Comment (champtar) [inline charts/metallb/templates/controller.yaml:52]: Not required anymore (remove it)

File: charts/metallb/templates/controller.yaml
Comment (champtar) [inline charts/metallb/templates/controller.yaml:58]: `METALLB_DEPLOYMENT` is not needed when memberlist is disabled, can you move the if/end accordingly ?

File: charts/metallb/templates/psp.yaml
Comment (champtar) [reply charts/metallb/templates/psp.yaml:86]: #764 is merged now

File: charts/metallb/templates/psp.yaml
Comment (champtar) [inline charts/metallb/templates/psp.yaml:64]: see https://github.com/metallb/metallb/pull/770/files, memberlist port has been added

File: charts/metallb/templates/speaker.yaml
Comment (champtar) [inline charts/metallb/templates/speaker.yaml:79]: see https://github.com/metallb/metallb/pull/770/files, memberlist port has been added

File: charts/metallb/templates/speaker.yaml
Comment (champtar) [inline charts/metallb/templates/speaker.yaml:50]: to be removed

**Review by champtar — CHANGES_REQUESTED** [review]
If you can do a diff with what is in master

File: charts/metallb/Chart.yaml
Comment (champtar) [inline charts/metallb/Chart.yaml:22]: 0.9.6 or 0.10.0 ?

File: charts/metallb/README.md
Comment (champtar) [inline charts/metallb/README.md:5]: 0.9.6 or 0.10.0 ?

File: charts/metallb/templates/psp.yaml
Comment (champtar) [inline charts/metallb/templates/psp.yaml:67]: you need to set the default value for `mlBindPort` else this is going to fail

File: charts/metallb/templates/speaker.yaml
Comment (champtar) [inline charts/metallb/templates/speaker.yaml:61]: As you are going to have a default `mlBindPort` let's make that `if .Values.speaker.memberlist.enabled`

File: charts/metallb/templates/psp.yaml
Comment (champtar) [reply charts/metallb/templates/psp.yaml:67]: Also what about `Values.speaker.memberlist.mlBindPort` ?

**Review by champtar — CHANGES_REQUESTED** [review]
Fix needed for `mlBindPort`

File: charts/metallb/templates/speaker.yaml
Comment (champtar) [reply charts/metallb/templates/speaker.yaml:61]: just saw it's in a `if .Values.speaker.memberlist.enabled`, so just remove it

File: charts/metallb/Chart.yaml
Comment (gclawes) [reply charts/metallb/Chart.yaml:22]: It should be whatever the latest release tag is.  It'll get updated by the pipeline when a new tag is cut.

Comment (gclawes) [reply charts/metallb/Chart.yaml:22]: I'll update it to 0.9.6

File: charts/metallb/README.md
Comment (gclawes) [reply charts/metallb/README.md:5]: 0.9.6

File: charts/metallb/templates/speaker.yaml
Comment (gclawes) [reply charts/metallb/templates/speaker.yaml:61]: Will remove.

File: charts/metallb/templates/speaker.yaml
Comment (champtar) [inline charts/metallb/templates/speaker.yaml:63]:
```suggestion
        - name: METALLB_ML_BIND_PORT
          value: "{{ .Values.speaker.memberlist.mlBindPort }}"
```

File: charts/metallb/templates/controller.yaml
Comment (champtar) [inline charts/metallb/templates/controller.yaml:48]: should the memberlist option be global ? it's a bit weird to have a speaker option impact the controller, even more so if we only want to deploy the controller (Calico BGP peering + MetalLB controller)

**Review by champtar — CHANGES_REQUESTED** [review]
I think those are the last 2 comments

**Comment by champtar** [comment]
Migration from bitnami to our chart works in my simple test
```
helm repo add bitnami https://charts.bitnami.com/bitnami

kubectl create namespace metallb-system

kubectl apply -f - <<'EOM'
apiVersion: v1
kind: ConfigMap
metadata:
  namespace: metallb-system
  name: config
data:
  config: |
    address-pools:
    - name: ingress-pool
      protocol: layer2
      addresses:
      - 10.10.52.150/32
      auto-assign: false
EOM

helm install metallb --namespace metallb-system bitnami/metallb --version 2.3.5 --set existingConfigMap=config

helm upgrade metallb --namespace metallb-system /root/metallb/charts/metallb --version 0.9.6 --values ~/metallbval.yaml
```

```
# cat ~/metallbval.yaml

existingConfigMap: config

controller:
  image:
    tag: main
    pullPolicy: Always

speaker:
  image:
    tag: main
    pullPolicy: Always
```

File: charts/metallb/templates/controller.yaml
Comment (champtar) [reply charts/metallb/templates/controller.yaml]: We have now switched to `{{- if and .Values.speaker.enabled .Values.speaker.memberlist.enabled }}`
@russellb except if you think we should move memberlist config to be at the top level just to be safe (if we find more memberlist config that are global) you can resolve this one

Comment (russellb) [reply charts/metallb/templates/controller.yaml]: I'll leave this to you both to decide.  I think it makes sense to think of memberlist as a speaker feature.  That's where it operates, right?  It's also probably harmless to set these env vars even if memberlist is not enabled.  Then you won't be looking at the setting from the controller stuff here

**Comment by champtar** [comment]
This magicaly triggered a metallb-0.9.6 "release"
image

**Comment by champtar** [comment]
this is latest commit from main, not v0.9.6

**Pull request merged**
